.github/workflows/gh-pages.yml

```diff
@@ -19,8 +19,6 @@ jobs:
     steps:
       - name:
         uses: actions/checkout@v4
-        with:
-          ref: main
```

Comment:
Why did you remove this condition?

Reply:
This forces CI to always check out the main branch instead of the branch for this PR.
This results in "Error: requirements.txt is not found" because the file has been introduced by this PR.

Reply:
That makes sense. We should check out the feature branch for pull-request events.